src/lib/statistics/statistics_objects/abstract_histogram.cpp

```diff
@@ -0,0 +1,1035 @@
+
+      GenericHistogramBuilder<T> builder{new_bin_count, _domain};
+
+      builder.add_copied_bins(*this, BinID{0}, value_bin_id);
+
+      // Do not create empty bin, if `value` is the only value in the bin
+      if (minimum != maximum) {
+        // A bin [50, 60] sliced with `!= 60` becomes [50, 59]
+        // TODO(anybody) Implement bin bounds trimming for strings
+        if constexpr (!std::is_same_v<
+                          pmr_string,
+                          T>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+          if (minimum == *value) {
+            minimum = _domain.next_value(*value);
+          }
+
+          if (maximum == *value) {
+            maximum = _domain.previous_value(*value);
+          }
+        }
+
+        const auto estimate = estimate_cardinality_and_distinct_count(PredicateCondition::Equals, variant_value);
+        const auto new_height = bin_height(value_bin_id) - estimate.cardinality;
+        const auto new_distinct_count = distinct_count - estimate.distinct_count;
+
+        builder.add_bin(minimum, maximum, new_height, new_distinct_count);
+      }
+
+      builder.add_copied_bins(*this, value_bin_id + 1, bin_count());
+
+      return builder.build();
+    }
+
+    case PredicateCondition::LessThanEquals:
+      return sliced(PredicateCondition::LessThan, _domain.next_value(*value));
+
+    case PredicateCondition::LessThan: {
+      auto last_bin_id = _bin_for_value(*value);
+
+      if (last_bin_id == INVALID_BIN_ID) {
+        last_bin_id = _next_bin_for_value(*value);
+
+        if (last_bin_id == INVALID_BIN_ID) {
+          last_bin_id = bin_count() - 1;
+        } else {
+          last_bin_id = last_bin_id - 1;
+        }
+      }
+
+      if (predicate_condition == PredicateCondition::LessThan && *value == bin_minimum(last_bin_id)) {
+        --last_bin_id;
+      }
+
+      auto last_bin_maximum = T{};
+      // previous_value(value) is not available for strings, but we do not expect it to make a big difference.
+      // TODO(anybody) Correctly implement bin bounds trimming for strings
+      if constexpr (
+          !std::is_same_v<
+              T, pmr_string>) {  // NOLINT clang-tidy is crazy and sees a "potentially unintended semicolon" here...
+        last_bin_maximum = std::min(bin_maximum(last_bin_id), _domain.previous_value(*value));
+      } else {
+        last_bin_maximum = std::min(bin_maximum(last_bin_id), *value);
+      }
+
+      GenericHistogramBuilder<T> builder{last_bin_id + 1, _domain};
+      builder.add_copied_bins(*this, BinID{0}, last_bin_id);
+      builder.add_sliced_bin(*this, last_bin_id, bin_minimum(last_bin_id), last_bin_maximum);
+
+      return builder.build();
+    }
+
+    case PredicateCondition::GreaterThan:
+      return sliced(PredicateCondition::GreaterThanEquals, _domain.next_value(*value));
+
+    case PredicateCondition::GreaterThanEquals: {
```

Comment:
Why is this less complex than LessThan?

Reply:
_next_bin_for_value can return INVALID_BIN_ID for LessThan, for GreaterThanEquals, this is already handled by does_not_contain, e.g.